sermant-plugins/sermant-mq-grayscale/mq-grayscale-rocketmq-plugin/src/main/java/io/sermant/mq/grayscale/interceptor/RocketMqAbstractIntercepter.java

```diff
@@ -0,0 +1,86 @@
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.mq.grayscale.interceptor;
+
+import io.sermant.core.plugin.agent.entity.ExecuteContext;
+import io.sermant.core.plugin.agent.interceptor.AbstractInterceptor;
+import io.sermant.core.utils.ReflectUtils;
+import io.sermant.core.utils.StringUtils;
+import io.sermant.mq.grayscale.service.RocketMqConsumerGroupAutoCheck;
+import io.sermant.mq.grayscale.utils.RocketMqGrayscaleConfigUtils;
+import io.sermant.mq.grayscale.utils.RocketMqSubscriptionDataUtils;
+
+import java.lang.reflect.Method;
+import java.util.Optional;
+
+/**
+ * mq consumer abstract interceptor
+ *
+ * @author chengyouling
+ * @since 2024-09-04
+ **/
+public class RocketMqAbstractIntercepter extends AbstractInterceptor {
+    @Override
+    public ExecuteContext before(ExecuteContext context) throws Exception {
+        return context;
+    }
+
+    @Override
+    public ExecuteContext after(ExecuteContext context) throws Exception {
+        if (!RocketMqGrayscaleConfigUtils.isPluginEnabled()) {
+            return context;
+        }
+        return doAfter(context);
+    }
+
+    /**
+     * Handling after the intercept point
+     *
+     * @param context context
+     * @return context
+     * @throws Exception Exception
+     */
+    protected ExecuteContext doAfter(ExecuteContext context) throws Exception {
+        return context;
+    }
+
+    /**
+     * build gray group name and set consumer client configs
+     *
+     * @param consumer consumer
+     * @param namesrvAddr namesrvAddr
+     * @param topic topic
+     * @param consumerGroup consumerGroup
+     * @throws Exception Exception
+     */
+    protected void buildGroupAndClientConfig(Object consumer, String namesrvAddr, String topic, String consumerGroup)
+            throws Exception {
+        String grayGroupTag = RocketMqGrayscaleConfigUtils.getGrayGroupTag();
+        if (StringUtils.isEmpty(grayGroupTag)) {
+            RocketMqConsumerGroupAutoCheck.setConsumerClientConfig(namesrvAddr, topic, consumerGroup);
+            return;
+        }
+        String grayConsumerGroup
+                = consumerGroup.contains("_" + grayGroupTag) ? consumerGroup : consumerGroup + "_" + grayGroupTag;
+        RocketMqSubscriptionDataUtils.setGrayGroupTagChangeMap(namesrvAddr, topic, grayConsumerGroup, true);
+        Optional<Method> method = ReflectUtils.findMethod(consumer.getClass(), "setConsumerGroup",
+                new Class[]{String.class});
+        if (method.isPresent()) {
+            method.get().invoke(consumer, grayConsumerGroup);
+        }
```

Comment:
this part can be not included in extrcted method for not using reflection

Reply:
fixed